api/ttn/lorawan/v3/devicerepository.proto

```diff
@@ -476,6 +484,24 @@ message MessagePayloadEncoder {
   repeated Example examples = 4 [(validate.rules).repeated.max_items = 20];
 }
 
+message EndDeviceProvisioningInfo {
+  EndDeviceModel end_device_model = 1;
+  EndDeviceTemplate end_device_template = 2;
+}
+
+message VendorProfile {
+  // ID of the LoRaWAN end device profile assigned by the vendor.
+  uint32 vendor_profile_id = 1;
+  // Device model identifier, as defined in the Device Repository.
+  string end_device_id = 2;
+  // Firmware version string.
+  string firmware_version = 3;
+  // Hardware version string.
+  string hardware_version = 4;
+  // Region.
+  string region = 5;
+}
+
```

Comment:
I removed the message. I also had to move the EndDeviceProfile struct in the store package because it's used by both the implementations of the store. There was already the old EndDeviceProfile struct, so I named the new one EndDeviceProfileV2.

I didn't modify the existing structure because none of the fields from the EndDeviceProfileV2 are common to the EndDeviceProfile. It didn't make sense to me to drag along all those other fields when none of them are needed for the provisioning info. I also renamed all the methods so it's more suggestive that they use the new structure. I'm not really sure if it's the right approach, let me know if there is a better way of doing this.